dev_scripts/qa.py

```diff
@@ -0,0 +1,448 @@
+        self.run_tests()
+        self.build_package()
+        self.build_qa_image()
+        self.qa_scenarios(skip=[1, 2, 8])
+
+
+class QADebianBased(QALinux):
+    """Base class for Debian-based distros.
+
+    This class simply points to the proper build instructions, and builds the Debian
+    package.
+    """
+
+    REF_BUILD = (
+        "https://github.com/freedomofpress/dangerzone/blob/main/BUILD.md#debianubuntu"
+    )
+
+    @QABase.auto("Build .deb", ref=REF_BUILD)
+    def build_package(self):
+        self.poetry_run(
+            # FIXME: Is installing stdeb required?
+            "pip",
+            "install",
+            "stdeb",
+            "&&",
+            "./install/linux/build-deb.py",
+        )
+
+
+class QAUbuntu2204(QADebianBased):
+
+    DISTRO = "ubuntu"
+    VERSION = "22.04"
+
+
+class QAUbuntu2210(QADebianBased):
+
+    DISTRO = "ubuntu"
+    VERSION = "22.10"
+
+
+class QAFedora(QALinux):
+    """Base class for Fedora distros.
+
+    This class simply points to the proper build instructions, and builds the RPM
+    package.
+    """
+
+    DISTRO = "fedora"
+    REF_BUILD = "https://github.com/freedomofpress/dangerzone/blob/main/BUILD.md#fedora"
+
+    @QABase.auto("Build .rpm", ref=REF_BUILD)
+    def build_package(self):
+        self.container_run(
+            "./dangerzone/install/linux/build-rpm.py",
+        )
+
+
+class QAFedora36(QAFedora):
+
+    VERSION = "36"
+
+
+class QAFedora37(QAFedora):
+
+    VERSION = "37"
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        prog=sys.argv[0],
+        description="Run QA tests for a platform",
+    )
+    parser.add_argument(
+        "platform",
+        choices=[p.value for p in Platform],
+        help="Run QA tests for the provided platform",
+    )
+    parser.add_argument(
+        "--try-auto",
+        action="store_true",
+        default=False,
+        help="Try to run the automated parts of the QA steps",
+    )
+    # FIXME: Find a better way to skip tasks in non-interactive environments, while
+    # retaining the ability to be semi-prompted in interactive environments.
+    parser.add_argument(
+        "--skip-manual",
+        action="store_true",
+        default=False,
+        help="Skip the manual parts of the QA steps",
+    )
+    parser.add_argument(
+        "--debug",
+        action="store_true",
+        default=False,
+        help="Enable debug logs",
+    )
+    return parser.parse_args()
+
+
+def setup_logging(debug=False):
+    """Simple way to setup logging.
+
+    Copied from: https://docs.python.org/3/howto/logging.html
+    """
+    # create logger
+    if debug:
+        lvl = logging.DEBUG
+    else:
+        lvl = logging.INFO
+
+    logger.setLevel(lvl)
+
+    # create console handler and set level to debug
+    ch = logging.StreamHandler()
+    ch.setLevel(lvl)
+
+    # create formatter
+    formatter = logging.Formatter("%(asctime)s - %(levelname)s - %(message)s")
```

Comment:
I didn't want to introduce colors right now, because we need special logic when piping the output of this script to a file, stuff that `git log` for instance does automatically. But I'd definitely want to introduce them at some point for visibility.

As for timestamps, I wanted to include them for two reasons:

1. It's an easy way to see how long each section has taken. Some steps (like building an image) take some time. A slow or fast execution may be a sign of something buggy.
2. It makes stdout / stderr from internal processes stand out, since they typically don't have timestamps.

Does this make sense?

Reply:
Sure, no need to add them now. And we can keep the time, but maybe a shorter version so it's not a verbose?